plugin/pkg/shoot/validator/admission.go

```diff
@@ -219,77 +271,79 @@ func (v *ValidateShoot) Admit(ctx context.Context, a admission.Attributes, o adm
 		// project name. These checks should only be performed for CREATE operations (we do not want to reject changes to existing
 		// Shoots in case the limits are changed in the future).
 		var lengthLimit = 21
-		if len(shoot.Name) == 0 && len(shoot.GenerateName) > 0 {
+		if len(c.shoot.Name) == 0 && len(c.shoot.GenerateName) > 0 {
 			var randomLength = 5
-			if len(project.Name+shoot.GenerateName) > lengthLimit-randomLength {
-				return apierrors.NewBadRequest(fmt.Sprintf("the length of the shoot generateName and the project name must not exceed %d characters (project: %s; shoot with generateName: %s)", lengthLimit-randomLength, project.Name, shoot.GenerateName))
+			if len(c.project.Name+c.shoot.GenerateName) > lengthLimit-randomLength {
+				return apierrors.NewBadRequest(fmt.Sprintf("the length of the shoot generateName and the project name must not exceed %d characters (project: %s; shoot with generateName: %s)", lengthLimit-randomLength, c.project.Name, c.shoot.GenerateName))
 			}
 		} else {
-			if len(project.Name+shoot.Name) > lengthLimit {
-				return apierrors.NewBadRequest(fmt.Sprintf("the length of the shoot name and the project name must not exceed %d characters (project: %s; shoot: %s)", lengthLimit, project.Name, shoot.Name))
+			if len(c.project.Name+c.shoot.Name) > lengthLimit {
+				return apierrors.NewBadRequest(fmt.Sprintf("the length of the shoot name and the project name must not exceed %d characters (project: %s; shoot: %s)", lengthLimit, c.project.Name, c.shoot.Name))
 			}
 		}
-		if strings.Contains(project.Name, "--") {
-			return apierrors.NewBadRequest(fmt.Sprintf("the project name must not contain two consecutive hyphens (project: %s)", project.Name))
-		}
-		// We don't want new Shoots to be created in Projects which were already marked for deletion.
-		if project.DeletionTimestamp != nil {
-			return admission.NewForbidden(a, fmt.Errorf("cannot create shoot '%s' in project '%s' already marked for deletion", shoot.Name, project.Name))
+		// TODO: move this to static project validation?
```

Comment:
WDYT?

Reply:
We should already have it: https://github.com/gardener/gardener/blob/9b21bbb67642f31f654448b4b2f5c1162c9066fa/pkg/apis/core/validation/project.go#L41 👀

Reply:
Right. Dropped the check in the admission plugin in favor of static validation.